scalding-serialization/src/main/scala/com/twitter/scalding/serialization/OrderedSerialization.scala

```diff
@@ -101,6 +101,30 @@ object OrderedSerialization {
     case NonFatal(e) => CompareFailure(e)
   }
 
+  def viaTransform[T, U](
```

Comment:
what about a variant that takes `U => Try[T]` for the deserialization side? For instance, cases where you serialize to strings (Json for instance), but clearly that can sometimes fail if we are reading random strings.